Makefile

```diff
@@ -1,18 +1,41 @@
-.PHONY: dev clean e2e push
+.PHONY: clean dev dev-private-cluster update-image-on-deployment push-tester-image deploy-e2e run-e2e
+
+# keep separate for simultaneous public/private dev without need for resource recreation
+clean-public:
+	rm -rf devenv/state devenv/public_cluster_tf/.terraform.lock.hcl devenv/public_cluster_tf/.terraform devenv/public_cluster_tf/terraform.tfstate devenv/public_cluster_tf/terraform.tfstate.backup
+
+clean-private:
+	rm -rf devenv/state devenv/private_cluster_tf/.terraform.lock.hcl devenv/private_cluster_tf/.terraform devenv/private_cluster_tf/terraform.tfstate devenv/private_cluster_tf/terraform.tfstate.backup
 
 dev:
 	terraform --version
-	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve
+	cd devenv && mkdir -p state && cd public_cluster_tf && terraform init && terraform apply -auto-approve
 
-clean:
-	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
-
-e2e:
-	go test -v --count=1 --tags=e2e ./e2e
+dev-private-cluster:
+	terraform --version
+	cd devenv && mkdir -p state && cd private_cluster_tf && terraform init && terraform apply -auto-approve
+	cd devenv && /usr/bin/env sh scripts/deploy_addon_private_cluster.sh
 
-push:
+# aka make push (formerly known as make push-private)
+update-image-on-deployment:
```

Comment:
the way we actually use this is becoming a lot more complicated. comments or docs explaining how to use these would be good.

this is becoming a lot more complicated than just 
make dev
make push
make e2e

Reply:
yep good call! I'll add a README